test/test_json_query_parser.cpp

```diff
@@ -0,0 +1,468 @@
+json long_lt = {{"kind", "lt"}, {"left", long_prop}, {"right", long_const}};
+json long_lte = {{"kind", "lte"}, {"left", long_prop}, {"right", long_const}};
+
+// double comparisons
+json double_eq = {{"kind", "eq"}, {"left", double_prop}, {"right", double_const}};
+json double_neq = {{"kind", "neq"}, {"left", double_prop}, {"right", double_const}};
+json double_gt = {{"kind", "gt"}, {"left", double_prop}, {"right", double_const}};
+json double_gte = {{"kind", "gte"}, {"left", double_prop}, {"right", double_const}};
+json double_lt = {{"kind", "lt"}, {"left", double_prop}, {"right", double_const}};
+json double_lte = {{"kind", "lte"}, {"left", double_prop}, {"right", double_const}};
+
+// null comparisons
+json int_null_eq = {{"kind", "eq"}, {"left", int_prop}, {"right", int_null_const}};
+json string_null_eq = {{"kind", "eq"}, {"left", string_prop}, {"right", string_null_const}};
+json float_null_eq = {{"kind", "eq"}, {"left", float_prop}, {"right", float_null_const}};
+json long_null_eq = {{"kind", "eq"}, {"left", long_prop}, {"right", long_null_const}};
+json double_null_eq = {{"kind", "eq"}, {"left", double_prop}, {"right", double_null_const}};
+
+// commutative expressions
+json int_commutative_eq = {{"kind", "eq"}, {"left", int_const}, {"right", int_prop}};
+json float_commutative_eq = {{"kind", "eq"}, {"left", float_const}, {"right", float_prop}};
+json string_commutative_eq = {{"kind", "eq"}, {"left", string_const}, {"right", string_prop}};
+json int_commutative_neq = {{"kind", "neq"}, {"left", int_const}, {"right", int_prop}};
+json float_commutative_neq = {{"kind", "neq"}, {"left", float_const}, {"right", float_prop}};
+json string_commutative_neq = {{"kind", "neq"}, {"left", string_const}, {"right", string_prop}};
+
+// string operations
+json string_begins_with = {{"kind", "beginsWith"}, {"left", string_prop}, {"right", begins_with_const}};
+json string_ends_with = {{"kind", "endsWith"}, {"left", string_prop}, {"right", ends_with_const}};
+json string_contains = {{"kind", "contains"}, {"left", string_prop}, {"right", contains_const}};
+json string_like = {{"kind", "like"}, {"left", string_prop}, {"right", like_const}};
+
+// case insensitive ops
+json string_begins_with_non_case = {
+    {"kind", "beginsWith"}, {"caseSensitivity", false}, {"left", string_prop}, {"right", begins_with_const_non_case}};
+json string_ends_with_non_case = {
+    {"kind", "endsWith"}, {"caseSensitivity", false}, {"left", string_prop}, {"right", ends_with_const_non_case}};
+json string_contains_non_case = {
+    {"kind", "contains"}, {"caseSensitivity", false}, {"left", string_prop}, {"right", contains_const_non_case}};
+json string_like_non_case = {
+    {"kind", "like"}, {"caseSensitivity", false}, {"left", string_prop}, {"right", like_const_non_case}};
+json string_eq_non_case = {
+    {"kind", "eqString"}, {"caseSensitivity", false}, {"left", string_prop}, {"right", string_const_non_case}};
+
+
+// sort operations
```

Comment:
like with sort, can you add some tests for distinct, and limit?

Reply:
Unfortunately not supported yet, added a new section in the document about supported features